aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteModelFieldTypeConverter.java

```diff
@@ -106,6 +152,11 @@ public Object convertValueFromSource(
                 case MODEL:
                     return convertModelAssociationToTarget(cursor, field);
                 case ENUM:
+                    @SuppressWarnings("rawtypes")
+                    final Class<? extends Enum> enumClass = field.getType().asSubclass(Enum.class);
+                    @SuppressWarnings({"rawtypes", "unchecked"})
+                    final Enum enumValue = Enum.valueOf(enumClass, valueAsString);
+                    return enumValue;
```

Comment:
`@SuppressWarnings({"rawtypes", "unchecked"})` can just go directly on the method itself to make it cleaner :)

Actually, rawtypes should be avoidable by doing this:
```
final Class<Enum<?>> enumClass = field.getType().asSubclass(Enum.class);
final Enum<?> enumValue = Enum.valueOf(enumClass, valueAsString);
```

Reply:
Actually, I'm not sure what the best way to represent `Class<T extends Enum<T>>` would be with a `?`, since `Class<? extends Enum<?>>` doesn't sufficiently enforce the two types to be the same... hmm

Reply:
Okay, my suggested solution doesn't work, and it seems that the only way to avoid `rawtypes` warnings is to do the following:
```
final Class<?> clazz = field.getType();
final Enum<?> enumValue = Enum.valueOf(clazz.asSubclass(Enum.class), "FOO");
```

Unchecked type warning is unfortunately very difficult to avoid, so you will need the annotation anyways :/

Reply:
Meh. Yea, I wasn't able to get rid of one final `unchecked` warning.
```
    @Test
    public void testEnumConversion() {
        assertEquals(Foo.BAR, valueOf(Foo.class, Foo.BAR.name()));
    }

    private static <E extends Enum<E>> Object valueOf(Class<?> clazz, String fieldName) {
        @SuppressWarnings("unchecked")
        Class<E> enumClazz = (Class<E>) clazz.asSubclass(Enum.class);
        return Enum.valueOf(enumClazz, fieldName);
    }

    enum Foo {
        BAR,
    }
```